### PR TITLE
fix(wdio-cucumber-framework): filter cucumberFeaturesWithLineNumbers

### PR DESCRIPTION
**Pull request by johnp**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

Fixes https://github.com/webdriverio/webdriverio/issues/13560. It looks like the `cucumberFeaturesWithLineNumbers` filtering broke some time ago, possibly with https://github.com/webdriverio/webdriverio/pull/11010.

While `_specs` has already been filtered here, `cucumberFeaturesWithLineNumbers`, which just exists to retain the line numbers, still contains all specs as passed on the cmdline or in the config. Therefore, just passing this to cucumber will inevitably execute all specs/features in every worker. To fix this, we only retain those entries within `cucumberFeaturesWithLineNumbers`, for which we can find a fitting entry in the list of already filtered `_spec`s.

Note that `cucumberFeaturesWithLineNumbers` does not really mean "cucumber features with line numbers". It seems to actually mean: "all passed specs, when at least one of them has a line number, or whatever was placed into the `cucumberFeaturesWithLineNumbers` wdio config option."

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Polish (an improvement for an existing feature)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (improvements to the project's docs)
- [ ] Specification changes (updates to WebDriver command specifications)
- [ ] Internal updates (everything related to internal scripts, governance documentation and CI files)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
    - [x] Test code which verifies that the correct feature file is executed (fails before the fix) 
    - [x] Test code which verifies that only the correct scenarios are executed
- [ ] I have added the necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Backport Request

[//]: # (The current `main` branch is the development branch for WebdriverIO v9. If your change should be released to the current major version of WebdriverIO (v8), please raise another PR with the same changes against the `v8` branch.)

- [ ] This change is solely for `v9` and doesn't need to be back-ported
- [x] Back-ported PR at #13563

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

Not 100% sure filtering with `startsWith` works 100% of the time (not sure if the paths could, for example, be relative, in one array, and absolute in the other, but they seem to always be absolute in my testing so far). **edit: now with `path.resolve`**

I wish Cucumber's `loadSources` would just take take whole features instead of just `paths`, but apparently it doesn't.

I'm not entirely sure if the `cucumberFeaturesWithLineNumbers` wdio config option is normalized anywhere. I don't know why it even exists, given that you can just put the features into `specs`, where the line numbers will get filtered out automatically.... It feels like `cucumberFeaturesWithLineNumbers` ought to be internal only.

### CODE REVIEW

File: infra/compiler/src/index.ts [hunk 1]
```diff
@@ -33,7 +34,7 @@ const { values } = parseArgs({ args, options })
  */
 const projects = (await fs.readdir(
     path.resolve(rootDir, 'packages')
-)).filter((dir) => dir !== 'node_modules')
+)).filter((dir) => dir !== 'node_modules' && existsSync(path.resolve(rootDir, 'packages', dir, 'package.json')))
```
Comment (christian-bromann) [inline infra/compiler/src/index.ts:37]: What is the origin of this change?

File: tests/cucumber/features/features.given.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { browser } from '@wdio/globals'
+import { Given } from '../../../packages/wdio-cucumber-framework/build/index.js'
```
Comment (christian-bromann) [inline tests/cucumber/features/features.given.js:2]: we use pnpm so we can just reference this as normal package
```suggestion
import { Given } from '@wdio/cucumber-framework'
```

**Review by christian-bromann — APPROVED** [review]
LGTM 👍 

Mind applying the same changes to the v8 branch?

**Comment by christian-bromann** [comment]
@johannespfrang I've been out on vacation for the last 3 weeks and am planning to release v7, v8 and v9 next week. We have others that could do releases as well but they don't have the context to all the changes which is why we prefer to have me around for them.

**Comment by aleks-temp** [comment]
And beside that, on version 9 when running from my local machine there is no problem with the multiple reruns, but there is the following issue on click:
WebDriver Bidi command "script.callFunction" failed with error: invalid argument - Invalid input in "arguments"/0

**Pull request merged**